### ISSUE TITLE
Shadow option is broken

### ISSUE THREAD

**Issue by davidjb99** [funk-team/funkLang#46]
@andressotomayorb commented on [Wed Mar 10 2021](https://github.com/funk-team/funk-enterprise/issues/690)

As a user when I try to add shadow to an element it is not reflected on the editor, but if I go to the preview, there it is visible.

https://user-images.githubusercontent.com/63779866/110684193-2301ad00-81dd-11eb-8973-bb3e3443de48.mov

**Comment by FranzSkuffka** [comment]
- [ ] @davidjb99 I suggest the outline disappears once the user adds a custom shadow because the element should then still be distinguishable from the background.

**Development Handoff**
The problem here is that funk uses shadows to draw outlines around elements. The user-defined shadows are overridden by the funk HUD-shadows.

*Relevant code sections*
- custom shadows are rendered here: https://github.com/funk-team/funkLang/blob/master/src/DesignSystem/Shadow.elm#L88
- shaodws are merged here: https://github.com/funk-team/funkLang/blob/master/src/Renderer/StyleCompositor.elm#L44
- shadows are incorrectly put into decorations rather than overriding existing shadows https://github.com/funk-team/funkLang/blob/master/src/Spec/Element/Render.elm#L315 which causes them to take a lower priority

*Proposed solutions*
- A: replace shadow rendering with renderer that outputs a `Maybe String`, then override the HUD shadows if we have `Just shadowString`
- B: Change shadows field type to `Element.Attribute msg`. Problem: this will not allow stacking multiple shadows and limit design options down the line


**Acceptance criteria**
- [ ] write e2e test that validates the behavior (shadow being rendered in `calculatedStyles`)
- [ ] fix the behavior

**Comment by FranzSkuffka** [comment]
Blocked by #50

**Comment by davidjb99** [comment]
I think hiding the shadow in the HUD as you suggest is fine when a user adds a shadow to an element. 

stacking shadows would be good to have at some point so I would go with option A
